event-based-simulators/main/event_based_simulators.py

```diff
@@ -386,22 +325,38 @@ def get_or_create_device_id(self):
                      'Piece_Quality': __on_piece_quality_event,
                      'Shutdown': __on_shutdown_event}
 
-    def __create_task(self, event_definition):
-        min_frequency_per_hour = event_definition.get("minimumFrequency", event_definition.get("frequency"))
-        max_frequency_per_hour = event_definition.get("maximumFrequency", event_definition.get("frequency"))
+    def __create_task(self, event_definition, measurement_definition):
+        if self.current_work == "event":
+            min_frequency_per_hour = event_definition.get("minimumFrequency", event_definition.get("frequency"))
+            max_frequency_per_hour = event_definition.get("maximumFrequency", event_definition.get("frequency"))
+
+            log.debug(f'create periodic task for {event_definition["type"]} ({min_frequency_per_hour}, {max_frequency_per_hour})')
+
+        elif self.current_work == "measurement":
+            min_frequency_per_hour = measurement_definition.get("minimumFrequency", measurement_definition.get("frequency"))
+            max_frequency_per_hour = measurement_definition.get("maximumFrequency", measurement_definition.get("frequency"))
+
+            log.debug(f'create periodic task for measurement {measurement_definition["series"]} with frequency in range ({min_frequency_per_hour}/hour, {max_frequency_per_hour}/hour)')
+
+        callback = MachineSimulator.lambda_functions(self, event_definition, measurement_definition)
 
         min_interval_in_seconds = int(3600 / max_frequency_per_hour)
         max_interval_in_seconds = int(3600 / min_frequency_per_hour)
 
-        event_callback = lambda task:  {MachineSimulator.event_mapping[event_definition["type"]](self, event_definition, task)}
+        task = PeriodicTask(min_interval_in_seconds, max_interval_in_seconds, callback)
 
-        task = PeriodicTask(min_interval_in_seconds, max_interval_in_seconds, event_callback)
-        
-        log.debug(f'create periodic task for {event_definition["type"]} ({min_frequency_per_hour}, {max_frequency_per_hour})')        
-        # event_callback()
         return task
-    
-    def __send_event(self, event_fragment, timestamp = None):
+
+    def lambda_functions(self, event_definition, measurement_definition):
+        if self.current_work == "event":
```

Comment:
use a constant for the strings "event" and "measurement" and use it everywhere where needed